aggregator/aggregator.go

```diff
@@ -361,48 +359,49 @@ func (agg *aggregator) updateShardsWithLock(newPlacement services.Placement) err
 	}
 
 	var (
-		newShards = instance.Shards()
-		incoming  []*aggregatorShard
-		closing   []*aggregatorShard
+		newShardSet = instance.Shards()
+		incoming    []*aggregatorShard
+		closing     = make([]*aggregatorShard, 0, len(agg.shardIDs))
 	)
 	for _, shard := range agg.shards {
 		if shard == nil {
 			continue
 		}
-		if !newShards.Contains(shard.ID()) {
+		if !newShardSet.Contains(shard.ID()) {
 			closing = append(closing, shard)
 		}
 	}
-	// NB(xichen): shard ids are guaranteed to be sorted in ascending order.
-	newShardIDs := newShards.AllIDs()
-	if len(newShardIDs) > 0 {
-		newShardLen := newShardIDs[len(newShardIDs)-1] + 1
-		incoming = make([]*aggregatorShard, newShardLen)
-	}
-	for _, shardID := range newShardIDs {
+
+	// NB(xichen): shards are guaranteed to be sorted by their ids in ascending order.
```

Comment:
What guarantees that the shards are sorted here? Does it make sense to be defensive here and sort the shards just in case? If the sorting is truly unnecessary perhaps add a comment explaining why?

Reply:
It's guaranteed by the `Shards()` implementation. I could create an issue in m3cluster to add comments saying the returned shards are always sorted by id in ascending order?

Reply:
Probably not necessary, maybe just augment the comment to note that `shards returned from instance.Shards() are guaranteed to be sorted ...`